aneeshrelan/node-multi-tenant-example
# feat(multi-tenancy): add wrapper for rabbitmq consumers, support for redis in middleware

Testing Redis

- Run  `./redis.sh` to create 20 tenants in redis container
- ``` curl -d '{"email":"tenant20", "taskData": [{"task": "task20"}]}' -H 'Content-Type: application/json' -X POST http://localhost:3000/users``` to add compay info in redis
- `curl -H "tenant-id: tenant20" localhost:3000/users` should return `tenantInfo` field

Testing RabbitMq
- `yarn pm2 start`
- `yarn pm2 logs`
- Go to `http://localhost:15672/` 
- Select Vhost (default is `deepak000000c14a5aa30c141efcc63v` ) This can be changed in `cache-multi-tenant/src/config/amqpChannel.ts` 
- Send `tenant1` in `assessmentWorker` and check logs ( Try with `tenant2` as well

Note: Have not updated the locust
